crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2466,16 +2482,138 @@ impl<'db> TypeInferenceBuilder<'db> {
         let ast::ExprCompare {
             range: _,
             left,
-            ops: _,
+            ops,
             comparators,
         } = compare;
 
         self.infer_expression(left);
-        // TODO actually handle ops and return correct type
-        for right in comparators.as_ref() {
-            self.infer_expression(right);
+        for expr in comparators {
+            self.infer_expression(expr);
+        }
```

Comment:
This should fix the MSRV build and also reduce the diff by 2 lines!

```suggestion
        for right in comparators.as_ref() {
            self.infer_expression(right);
```